tests/perp_test.py

```diff
@@ -0,0 +1,71 @@
+# perp_test.py
+from grpc._channel import _InactiveRpcError
+
+import tests
+from nibiru import common
+
+
+class TestPerp(tests.ModuleTest):
+    def setUp(self):
+        self.validator = tests.get_val_node(tests.get_network())
+        self.market = "ubtc:unusd"
+
+    def test_open_close_position(self):
```

Comment:
I also wrote a test for PERP open_position but with deeper test of the opened position values:
https://github.com/NibiruChain/nibiru-py/pull/65/files#diff-ee3290e1bcd93463a80e09f089548c88b369dc8bf4feb758acc7c2483407514e

Without a fix to position serialization/deserialization opened positions are kind of invalid.

Reply:
We can check the status of the position after the position is opened, but the perp tests will not work if there's already a position opened for this trader before. I can modify this one to ensure that position is closed before opening a new one

Reply:
Ideally test should be isolated enough from the other tests. So maybe we better open position for a newly created trader rather than validator. In that case position will only belong to the one session of the unit test.

Reply:
Yes but in that case he needs funds from somewhere, which can vary if we have or not a faucet. The best solution would be to create a localnet and make the tests from there

Reply:
As soon as it's kind of unit tests, I don't think we are going to execute it across anything but localnet (or devnet in this case which is localnet). So, assuming we have full control against that chain and we always have a validator available.

Reply:
@matthiasmatt Keep in mind that the chain is frequently restarted and that the tests should be reasonably expected to run whether it's a fresh or running chain. 

It needs to be written in such a way that if we ran it back to back, the tests would still run. It's also possible that other people are running workflows locally or in the cloud as well ([example E2E test - See "Run devnet tests" section](https://github.com/NibiruChain/localnet/runs/7829267875?check_suite_focus=true)). 

If all else fails, you can use the `--generate-only` flag to verify the initialization for the tx is correct without submitting it. 

There's also a `dryrun` flag on certain commands that preps the message and shos a gas estimate without submitting it to the chain.

Reply:
And yes, the validator, oracle, and liquidator are available. You can send them money or use `keys add` to get more complex behavior in a test.

Reply:
Ok! Adding this logic